README.md

```diff
@@ -148,6 +148,133 @@ You should now have the app running on <https://localhost:5000/>
 
+## Local development with docker
+
+The webapp depends on keycloak, a postgres instance holding metadata, an s3 bucket storing associated records and then an opensearch instance that is populated from those 2 via `data_management/opensearch_indexer`. For ease of use, we provide a `docker-compose.yml` file inside the `local_services` which spins up all these dependencies, using minio as a local replacement for an actual AWS s3, and populates them with consistent test data. Feel free to expand this data but data consistency is left up to you.
+
+There are some dependencies for running this docker compose stack.
+
+1. Have `docker` installed
+2. Create certs for the webapp postgres instance in `local_services/webapp_postgres_certs` by running `generate_webapp_postgres_certs.sh` inside it
+3. Create certs for the opensearch nodes in `local_services/opensearch_certs` by running `generate_opensearch_certs.sh` inside it
+4. Create a `.env` file inside of `local_services` using `local_services/.env.template`
+
+Then you can run: `docker compose up -d`
+
+It will take a minute or two to spin up the stack, in particular opensearch and keycloak take a little while. You can check their progress in each container's logs.
+
+Once the stack is running:
+
+1. Create your users in the keycloak admin console at `http://localhost:8080/admin/master/console/#/tdr/users` using the keycloak admin credentials specified in the `.env` file, assigning appropriate groups to each. For local dev it's simple enough to set the passwords in the credentials tab.
+2. Regenerate the keycloak client's client secret at `http://localhost:8080/admin/master/console/#/tdr/clients/<UUID-OF-CLIENT>/settings`.
+3. Update the `.env` of the webapp in the root directory of the repo, making sure the following env vars are set according to the values set for the associated service in the docker compose stack.
+
+```
+export KEYCLOAK_BASE_URI=http://localhost:8080
+export KEYCLOAK_REALM_NAME=tdr
+export KEYCLOAK_CLIENT_ID=ayr-beta
+export KEYCLOAK_CLIENT_SECRET=<secret regenerated above>
+
+
+export DB_PORT=5433
+export DB_HOST=localhost
+export DB_NAME=local_db
+export DB_USER=local_db_user
+export DB_PASSWORD=local_db_user_password
+
+
+export AWS_ENDPOINT_URL=http://localhost:9000
+export AWS_ACCESS_KEY_ID=ROOTNAME
+export AWS_SECRET_ACCESS_KEY=CHANGEME123
+
+export OPEN_SEARCH_HOST=https://localhost:9200
+export OPEN_SEARCH_USERNAME=admin
+export OPEN_SEARCH_PASSWORD=FOOBARCARabc123!
+export OPEN_SEARCH_CA_CERTS=local_services/opensearch_certs/root-ca.pem
+
+export DB_SSL_ROOT_CERTIFICATE=local_services/webapp_postgres_certs/root-ca.pem
+```
+
+Finally you can populate the opensearch cluster with the corresponding data stored in snapshot 1 in `local_services/snapshots/` by running `source .env && local_services/opensearch-entrypoint.sh`.
+
+Then you can run the flask server with `flask run`
+
+## Local dev without docker
```

Comment:
ignore this section, I just moved this from below and put it under "Local dev without docker" in case anyone would want to continue doing that